core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -762,14 +762,15 @@ public ReplicationPhase(ReplicaRequest replicaRequest, Response finalResponse, S
             // we have to make sure that every operation indexed into the primary after recovery start will also be replicated
             // to the recovery target. If we use an old cluster state, we may miss a relocation that has started since then.
             // If the index gets deleted after primary operation, we skip replication
-            final ClusterState state = clusterService.state();
+            ClusterState state = clusterService.state();
```

Comment:
That was an inadvertent loss; I pushed 42ec0fe504444c8eb122c6b05377fb2cb294cb6b.